### PR TITLE
Add assetStructure to dbt model views

### PR DESCRIPTION
**Pull request by usefulalgorithm**
### 🤔 Why?

To populate `AssetStructure` for DBT model views.

### 🤓 What?

- Added logic to keep track of model's directories and name.

### 🧪 Tested?

Unit test.

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: metaphor/dbt/artifact_parser.py [hunk 1]
```diff
@@ -452,6 +453,12 @@ def _parse_model(
             return
 
         virtual_view = init_virtual_view(self._virtual_views, model.unique_id)
+
+        virtual_view.structure = AssetStructure(
+            directories=model.original_file_path.split("/")[:-1],
```
Comment (mars-lan) [inline metaphor/dbt/artifact_parser.py:458]: For dbt, I believe we should only have a single-level directory, i.e. the project directory name.

Comment (usefulalgorithm) [reply metaphor/dbt/artifact_parser.py:458]: project directory name == model.path ?

I think it should be something else but can't figure out what

File: metaphor/dbt/artifact_parser.py [hunk 2]
```diff
@@ -455,8 +455,12 @@ def _parse_model(
 
         virtual_view = init_virtual_view(self._virtual_views, model.unique_id)
 
+        # Extract project directory from the model's unique id
+        directory = get_model_name_from_unique_id(model.unique_id)[
+            : -len(model.name) - 1
```
Comment (mars-lan) [inline metaphor/dbt/artifact_parser.py:460]: Consider split by `.` instead, which seems a bit more intuitive/robust?

**Review by mars-lan — COMMENTED** [review]
Otherwise LGTM.

**Review by mars-lan — APPROVED** [review]

**Review by mars-lan — APPROVED** [review]

**Review by alyiwang — COMMENTED** [review]
LGTM

**Pull request merged**